src/main/java/seedu/address/logic/parser/RemoveClassCommandParser.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLASS;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.RemoveClassCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.module.Class;
+import seedu.address.model.module.ClassName;
+
+/**
+ * Parses input from the user and creates a RemoveClassCommand.
+ */
+public class RemoveClassCommandParser implements Parser<RemoveClassCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the RemoveClassCommand
+     * and returns a RemoveClassCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RemoveClassCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CLASS);
+        if (!arePrefixesPresent(argMultimap, PREFIX_CLASS)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveClassCommand.MESSAGE_USAGE));
+        }
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_CLASS);
+        Class c = null;
+        try {
+            ClassName classname = ParserUtil.parseClassName(argMultimap.getValue(PREFIX_CLASS).get());
+            c = new Class(classname);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveClassCommand.MESSAGE_USAGE), pe);
+        }
+        return new RemoveClassCommand(c);
```

Comment:
I like that the code is neat and easy to understand but does this `RemoveClassCommand(c)`  work because I thought for me, it was more intuitive to do something like getting the Class c from the `UniqueClassList` from `Model Manager` and passing into `RemoveClassCommand`. 

If it works, please ignore! I might have missed out something in your implementation in that case 👍

Reply:
I think your way of implementation would work as well!
I was thinking more of
RemoveClassCommandParser -> Only parses the user input (Ensures that valid fields are there)
The part of whether the removal of Class should happen is handled by the lower level implementation (In my case, I let the UniqueClassList do it. Only when we execute the RemoveClassCommand) 
If we check whether the class is present inside the parser, imo this might be going against abstraction since the parser's job is just to make sure the user has valid input.